mage_ai/frontend/pages/pipelines/[pipeline]/triggers/index.tsx

```diff
@@ -197,25 +197,39 @@ function PipelineSchedules({
   const buildSidekick = useMemo(() => {
     const variablesOverride = selectedSchedule?.variables;
     const hasOverride = !isEmptyObject(variablesOverride);
-
-    const showVariables = hasOverride
-      ? selectedSchedule?.variables
-      : !isEmptyObject(variablesOrig) ? variablesOrig : null;
+    const hasVariables = !isEmptyObject(variablesOrig);
 
     return (props: DependencyGraphProps) => {
-      const dependencyGraphHeight = props.height - (showVariables ? 151 : 80);
+      /** 
+       * Because it's required to specify the DependencyGraph height, we calculate
+       * the RuntimeVariables height here instead of within its component.
+       * We dynamically calculate the RuntimeVariables height based on the number
+       * of visible rows in the runtime variables table at any time.
+       */
+      let runtimeVariablesHeight = 80;
+      if (hasVariables) {
+        const maxVisibleRows = 5;
+        const rowHeight = 44;
+        const numVariables = Object.keys(variablesOrig).length;
+        const numVisibleRows = Math.min(maxVisibleRows, numVariables);
+        // This accounts for title + spacing + table in RuntimeVariables
+        runtimeVariablesHeight = 116 + numVisibleRows * rowHeight;
+      }
+   
+      const dependencyGraphHeight = props.height - runtimeVariablesHeight;
```

Comment:
A bit hacky but height was already previously hardcoded, so this was a minimal solution I could think of without a much larger refactor.